libpod/runtime_ctr.go

```diff
@@ -50,6 +51,24 @@ func (r *Runtime) NewContainer(rSpec *spec.Spec, options ...CtrCreateOption) (c
 	ctr.state.State = ContainerStateConfigured
 	ctr.runtime = r
 
+	// Check CGroup parent sanity, and set it if it was not set
+	switch r.config.CgroupManager {
+	case CgroupfsCgroupsManager:
+		if ctr.config.CgroupParent == "" {
+			ctr.config.CgroupParent = CgroupfsDefaultCgroupParent
+		} else if strings.HasSuffix(path.Base(ctr.config.CgroupParent), ".slice") {
+			return nil, errors.Wrapf(ErrInvalidArg, "systemd slice received as cgroup parent when using cgroupfs")
+		}
+	case SystemdCgroupsManager:
+		if ctr.config.CgroupParent == "" {
+			ctr.config.CgroupParent = SystemdDefaultCgroupParent
+		} else if len(ctr.config.CgroupParent) < 6 || !strings.HasSuffix(path.Base(ctr.config.CgroupParent), ".slice") {
```

Comment:
I know just enough about cgroups to be dangerous.  Should ".slice" here be "system.slice"?  Could a cgroups.slice string satisfy this and I'm not sure you'd want that?  @rhatdan?

Reply:
I'm pretty sure that anything with .slice is systemd-managed, which is why we want to avoid them when using cgroupfs - but not 100%. For what it's worth, this code is basically identical to CRI-O's validation.

Reply:
Yes .slice means systemd.